lib/logmsg/logmsg.c

```diff
@@ -203,6 +203,25 @@ const gchar *builtin_value_names[] =
   NULL,
 };
 
+const gchar *
+log_msg_value_type_to_str(LogMessageValueType self)
+{
+  static const gchar *as_str[] = {
+    [LM_VT_STRING] = "STRING",
+    [LM_VT_JSON] = "JSON",
+    [LM_VT_BOOLEAN] = "BOOLEAN",
+    [LM_VT_INT32] = "INT32",
+    [LM_VT_INT64] = "INT64",
+    [LM_VT_DOUBLE] = "DOUBLE",
+    [LM_VT_DATETIME] = "DATETIME",
+    [LM_VT_LIST] = "LIST",
+    [LM_VT_NULL] = "NULL",
+    [LM_VT_NONE] = "NONE",
+  };
```

Comment:
Could you add an assert () to ensure that we don't access that array out of bounds?

Maybe a static assert would be sufficient.

Also I'd prefer lowercase names for the types, that way we would match type hints.

And one last comment, completely optional: we have the reverse mapping in the type hint module, maybe we could refactor both cases into one set of functions/array.

Reply:
Sure, I am looking into your points!

Reply:
Merging the two functions to use the same array was turning to be rather unpleasant, but I was able to add a `log_msg_value_type_from_str()` function for the sake of symmetry. If you are not fond of it, we can drop the last commit. :)

Reply:
This looks good to me. Thanks